### DISCUSSION TITLE
Workshop not starting

### DISCUSSION THREAD

**Discussion by martinmacecek-rearc** [noop-inc/documentation#130] [Questions]
Got a problem with the desktop app on Mac (0.9.0-arm64): after installation and signing in through Github, the Workshop seems to start, but remains in this state:
<img width="891" alt="Screenshot 2024-02-28 at 10 35 00 AM" src="https://github.com/noop-inc/documentation/assets/156843182/d2ccb062-9e8a-47a9-9b2e-04bb9855bb02">

I also cannot seem to be able to fully quit the application, which may be related to the workshop startup problem. I recorded this behavior: 
https://github.com/noop-inc/documentation/assets/156843182/9319e6f6-8ee2-4e24-bd7a-2a7048165df0

Now, when i force quit through Activity Monitor, I get this error on initial startup:
<img width="519" alt="Screenshot 2024-02-28 at 10 42 53 AM" src="https://github.com/noop-inc/documentation/assets/156843182/d117a5bb-bf28-460a-abe5-5e49c7c5ff72">

After that I am back to the workshop not successfully starting and the app being unable to quit without using force.

So, I went ahead and uninstalled the app and reinstalled, but i am immediately presented with the above error message again, which then, upon confirmation, brings me to the main screen, but the workshop again unable to start.

Any ideas of what to do to furhter troubleshoot/resolve this? I am not sure if I did something wrong or if this is truly a bug.

**Comment by dupuyarc** [comment]
As I wrote in https://github.com/orgs/noop-inc/discussions/120, a system restart cleared the problem for me. It's a bit heavy-handed, but it does the trick.

**Reply by arbitrarytech** [reply]
Were either of you running a prior test release like 0.8.2 before updating to 0.9.0 by chance? We're aware of issues in that release that can orphan processes and leave internal system artifacts that prevent subsequent executions of Workshop.

**Reply by martinmacecek-rearc** [reply]
For me, it was the first installation on my system.

**Reply by dupuyarc** [reply]
Yes, I did my initial testing yesterday with 0.8.2, and I only upgraded to 0.9.0 _after_ I was already encountering the problem with failure to start the Workshop.

**Reply by martinmacecek-rearc** [reply]
A restart did indeed do the trick for me as well. The situation is not ideal and the fact that the app does not quit made it even more confusing.
Is there at least an FAQ where this could be documented, as I am clearly not the first one running into that? Unless of course there is a solution to it.

**Reply by dupuyarc** [reply]
It seems like the system restart is the solution if you are having this problem already.

If you upgrade to 0.9.0 the problem "shouldn't happen"? At the very least, it hasn't happened for me, and quitting the app works without hanging (it does mistakenly display "starting Workshop" during the shutdown process, but that's a very short glitch and not really a problem).

**Reply by martinmacecek-rearc** [reply]
I started with `0.9.0`, so maybe that version does not cover all problematic situations, which lead to the workshop not starting.

**Comment by martinmacecek-rearc** [comment]
So ,the problem is back after I just tried to quit the app. Unfortunately the app does not quit gracefully for me and I had to force quit it and now I am experiencing the error above and the workshop not starting again. 

I don't think that restarting my computer each time i want to quit and later re-open the app is intended, so I unmarked the initial answer.
IMHO and in my case it all seems to start with the app being unable to quite gracefully.

Unfortunately I have not been able to reliably replicate the app being unable to quit. I thought it might have been related to an environment being active while i attempted to quit, but that wasnt the case in my follow up attempt after a restart.

**Reply by 1800joe** [reply]
Can you confirm this is still a problem in v0.9.1?

**Reply by martinmacecek-rearc** [reply]
I still don't have the exact circumstances, but I kept the app running for a while, without really doing anything. I then tried to quit and it would indicate the `Workshop stopped` (while still spinning the progress icon), but the app never quit. I then tried to quit again and the app quit successfully.
But when I start the app again and, immediately after the workshop is started, I quit, there is no such problem.
I can only suspect there is some process not being cleaned up/reporting its shutdown correctly after prolonged run time of the app that prevents the initial quit.

That all being said, I am not sure if you really want to prioritize this as a problem, as it seems that I at least no longer require to force quit the app to fully stop it with version `0.9.1`.
I also am not 100% certain on the prolonged runtime causing any issues, but it seems to be the pattern that emerged for me.
